### ISSUE TITLE
Makefile: no Clang found?

### ISSUE THREAD

**Issue by ghost** [Infinidat/infinitrace#3]
```
gcc -L.  trace_dumper/trace_dumper.o trace_dumper/filesystem.o trace_dumper/events.o trace_dumper/sgio_util.o trace_dumper/internal_buffer.o trace_dumper/mm_writer.o trace_dumper/writer.o trace_dumper/write_prep.o trace_dumper/buffers.o trace_dumper/init.o trace_dumper/open_close.o trace_dumper/metadata.o trace_dumper/housekeeping.o trace_user_stubs.o -pthread -ltrace -ltraceutil -lparser -lsnappy  -lrt -o trace_dumper/trace_dumper 
Makefile:77: No clang found, skipping instrumentor build
```

``` bash
constantine@linux /mnt/space/dev/infinitrace $ get install clang-3.3 llvm-3.3
Reading package lists... Done
Building dependency tree       
Reading state information... Done
clang-3.3 is already the newest version.
llvm-3.3 is already the newest version.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
constantine@linux ~ $ clang --version
Ubuntu clang version 3.3-16ubuntu1 (branches/release_33) (based on LLVM 3.3)
Target: x86_64-pc-linux-gnu
Thread model: posix
```

**Comment by ghost** [comment]
Found in Makefile these lines:

```
REQUIRED_CLANG_VER=3.3
CLANG_VER=$(shell which $(CLANG) > /dev/null && $(CLANG) --version|head -1|grep -oP '\d\.\d '|cut -d' ' -f1)
```

Testing it in shell:

```
constantine@linux ~ $ CLANG=clang
constantine@linux ~ $ CLANG_VER=$(which $CLANG > /dev/null && $CLANG --version|head -1|grep -oP '\d\.\d '|cut -d' ' -f1)
constantine@linux ~ $ echo $CLANG_VER

```

Prints nothing.

**Comment by ghost** [comment]
```
constantine@linux ~ $ $CLANG --version|head -1
Ubuntu clang version 3.3-16ubuntu1 (branches/release_33) (based on LLVM 3.3)
constantine@linux ~ $ $CLANG --version|head -1|grep -oP '\d\.\d '
constantine@linux ~ $
```

Seems like grep command is not working.

**Comment by ghost** [comment]
Bingo.

```
constantine@linux ~ $ $CLANG --version|head -1|grep -oP '\d\.\d'
3.3
3.3
```

There should be no extra space in expression.

Added: now I get two of them for some reason :)

```
Makefile:82: Wrong clang version 3.3 3.3 found. Version 3.3 is required, skipping instrumentor build
```

...
Oh, of course, LLVM 3.3 :) ...

**Comment by ghost** [comment]
This do the trick:

```
constantine@linux ~ $ $CLANG --version|head -1
Ubuntu clang version 3.3-16ubuntu1 (branches/release_33) (based on LLVM 3.3)
constantine@linux ~ $ $CLANG --version|head -1|grep -oP 'version \d\.\d'|cut -d' ' -f2
3.3
```

**Comment by baruch** [comment]
This was fixed in commit 31711a31a5273af940a4995cb591e445c3efd72e
